packages/amplify_auth_cognito/lib/method_channel_auth_cognito.dart

```diff
@@ -316,9 +314,7 @@ class AmplifyAuthCognitoMethodChannel extends AmplifyAuthCognito {
         nextStep: AuthNextSignInStep(
             signInStep: res["nextStep"]["signInStep"],
             codeDeliveryDetails: res["nextStep"]["codeDeliveryDetails"],
-            additionalInfo: res["nextStep"]["additionalInfo"] is String
-                ? jsonDecode(res["nextStep"]["additionalInfo"])
-                : {}));
+            additionalInfo: res["nextStep"]["additionalInfo"] ?? {}));
```

Comment:
It looks like Android and iOS just use codeDeliveryDetails.  We can probably back this change out for now, and remove these unused attributes later.

Reply:
`codeDeliveryDetails` would not work for custom auth flow, only with the out-of-the-box MFA cognito flow. The public challenge parameters can be anything, which can be unrelated to delivering a code (e.g. a Captcha flow).

Reply:
@haverchuck: codeDeliveryDetails would not work for public challenge parameters since those can be any set of key/values returned, not just specific to code delivery. For example, you might be returning a Captcha url.

This change is definitely needed for the fix. I did back out the change that I made to `_formatSignUpResponse`, since that's not needed for the fix. Was that the line you meant to refer to?

Reply:
I recalled that you had a change related to `additionalInfo` in the response from the resendSignUpCode method, but I might be mistaken (I'm not seeing it now).  These changes look ok to me.  I've asked our ios team to test your PR in the amplify-ios repo, and once they've released it I'll retest this PR a bit and then merge it.